docs/configuration/dynamic-config.md

```diff
@@ -0,0 +1,118 @@
+# Dynamic Configuration - BETA
+
+**This is BETA and subject to change at anytime, feedback is much appreciated**
+
+Dynamic Configuration is the combination of two things:
+
+* Loading from multiple files
+* Using templates and datasources
```

Comment:
Are these features going to be tightly coupled? e.g. will users be able to make use of multiple config files w/o enabling the dynamic config feature that parses the gomplate templates? In my mind these seem like they should be separate.

Reply:
The templating is optional, if not needed inside the config files then you don't include the template markup. I do think it is integral to the feature though, ie if they were separate PRs to keep the code review easier that's fine but releasing together is important.